server/src/internalClusterTest/java/org/opensearch/search/fields/SearchFieldsIT.java

```diff
@@ -862,6 +862,9 @@ public void testDocValueFields() throws Exception {
                 .startObject("ip_field")
                 .field("type", "ip")
                 .endObject()
+                .startObject("flat_object_field")
```

Comment:
This is great that you added the `flat_object` to document, but the tests do not use this field from the script, I would expect something along this lines:

```
{
  "query": {
    "bool": {
      "filter": {
        "script": {
          "script": "def title = doc['catalog.title'].value; title.size() > 10"
        }
      }
    }
  }
}
```

It does not work now and I am not sure how the `flat_object` field could be accessible from script.

Reply:
@reta I do not think it is currently possible to access internal fields of the `flat_object` field from within filter scripting. But at least it is possible to filter documents based on number of the fields. I added test for this.

Reply:
Sure, @mingshl could you please add it to the list of limitations (and follow improvements), thank you